src/par_iter/collect/mod.rs

```diff
@@ -0,0 +1,38 @@
+use super::ExactParallelIterator;
+use std::isize;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+mod consumer;
+use self::consumer::CollectConsumer;
+
+pub fn collect_into<PAR_ITER,T>(mut pi: PAR_ITER, v: &mut Vec<T>)
+    where PAR_ITER: ExactParallelIterator<Item=T>,
+          PAR_ITER: ExactParallelIterator,
+          T: Send,
+{
+    let len = pi.len();
+    assert!(len < (isize::MAX) as usize);
+
+    v.truncate(0); // clear any old data
+    v.reserve(len); // reserve enough space
+    let target = v.as_mut_ptr(); // get a raw ptr
+    let writes = AtomicUsize::new(0);
+    let consumer = unsafe { // assert that target..target+len is unique and writable
+        CollectConsumer::new(&writes, target, len)
+    };
+    pi.drive(consumer);
```

Comment:
Hmm, interesting thought! That would indeed simplify a lot. I have a few reservations though:

1. I've been planning to extend `collect` to work for bounded parallel iterations (e.g., with filter). The idea would be to track where writes have occurred and then do a compression step (if needed). But then again, I guess that would work fine with your approach.

2. I wanted to drop partially initialized results on panic. This would be fairly easy with the current approach (when the "Folder" gets dropped, if it hasn't finished, it can drop those values its written thus far; similarly, it can return a dummy value as its Result. When this value is reduced, it's a no-op, but if it gets dropped, it drops the completed values). However, if we adopt the "track where writes occurred" approach, that can be handled too.

3. We're relying on various bits of safe code, but it's all safe code we control I suppose, so that seems ok.

Hmm so maybe it would work out. That'd be very cool.